sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/DataLakeFileAppendOptions.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.options;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.CoreUtils;
+
+/**
+ * Optional parameters for appending data to a file.
+ */
+@Fluent
+public class DataLakeFileAppendOptions {
+
+    private String leaseId;
+    private byte[] contentHash;
+    private Boolean flush;
+
+
+    /**
+     * Gets the lease ID to access the file.
+     *
+     * @return lease ID to access this file.
+     */
+    public String getLeaseId() {
+        return leaseId;
+    }
+
+    /**
+     * Sets the lease ID.
+     *
+     * @param leaseId The lease ID.
+     * @return the updated DataLakeFileAppendOptions object.
+     */
+    public DataLakeFileAppendOptions setLeaseId(String leaseId) {
+        this.leaseId = leaseId;
+        return this;
+    }
+
+    /**
+     * When this header is specified, the storage service compares the hash of the content that has arrived with this
+     * header value. If the two hashes do not match, the operation will fail with error code 400 (Bad Request). Note
+     * that this MD5 hash is not stored with the file. This header is associated with the request content, and not with
+     * the stored content of the file itself.
+     *
+     * @return MD5 hash of the content of the data.
+     */
+    public byte[] getContentHash() {
```

Comment:
Any reason this was decided to be `contentHash` instead of `contentMd5` as used in other location?

Reply:
.NET calls their method contentHash, but I will change to contentMd5 since it's what we use everywhere else in Java.

Reply:
It is `ContentHash` in .NET, and I'm not opposed to changing it to `ContentHash` here, but I think it's important for each API to be internally consistent.  This is a good question, might be something for us to discuss in the Arch Board meeting.  @ibrahimrabab, when we generate the API views, please leave a comment on `DataLakeFileAppendOptions.getContentMd5()` linking to this discussion.